### PR TITLE
Site Editor: Export Editor component

### PR DESCRIPTION
**Pull request by ockham**
## Description
Export the Editor component from the `@wordpress/edit-site` package. This will allow us to embed that component in different JS apps.

## How has this been tested?
Doesn't really need testing.

## Types of changes
New export.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by noahtallen — APPROVED** [review]
This looks fine to me. Maybe should also get a ✅ from @youknowriad

**Comment by ockham** [comment]
cc/ @saramarcondes

**Comment by youknowriad** [comment]
Can you clarify more how is this package used in alternative JS apps. If we're making this an API we need to understand how to maintain backward compatibility for it. 

In general we consider `edit-*` packages as WP specific packages not tailored to be used outside WP-Admin.

File: packages/edit-site/src/index.js [hunk 1]
```diff
@@ -64,4 +64,5 @@ export function initialize( id, settings ) {
 	render( <Editor />, document.getElementById( id ) );
 }
 
+export { Editor };
```
Comment (ntsekouras) [inline packages/edit-site/src/index.js:67]: Why not just import from `components/editor` where needed?

Comment (ntsekouras) [reply packages/edit-site/src/index.js:67]: Riad explained it to me. :)

**Comment by saramarcondes** [comment]
@ockham @youknowriad is right. I'm running into this currently trying to use `edit-post` in this PR: https://github.com/Automattic/wp-calypso/pull/44801. The usage of `@wordpress/core-data` is pretty problematic. To actually enable the use of the `Editor` component in other JS apps (outside of WP admin) we'd need to replace the use of `core-data` with something configurable through props to the `Editor` component (there's probably a handful of different ways to do this I would imagine).

**Comment by ockham** [comment]
@saramarcondes Oh, did my suggestion about using `apiFetch` middlewares and custom fetch handlers to intercept and modify `core-data`'s network requests not work?

**Comment by saramarcondes** [comment]
@ockham Ah, I haven't had a change to try that out yet, I didn't realize that it could be a wholesale solution to this but now I'm realizing what you mean.

Do you think that's the best approach to being able to use these components outside of wp admin though? I think I'd prefer actually spending some time to develop an API for it rather than doing something that is admittedly kind of a hack. It seems like it would be a nightmare to maintain longterm and even less intuitive/more stuff to explain to a new dev coming into the area.

**Comment by ockham** [comment]
> @ockham Ah, I haven't had a change to try that out yet, I didn't realize that it could be a wholesale solution to this but now I'm realizing what you mean.
> 
> Do you think that's the best approach to being able to use these components outside of wp admin though? I think I'd prefer actually spending some time to develop an API for it rather than doing something that is admittedly kind of a hack. It seems like it would be a nightmare to maintain longterm and even less intuitive/more stuff to explain to a new dev coming into the area.

Yeah, that's a fair point. My main goal here is to control the `settings` we pass to the editor, so we can apply a wash-rinse-repeat approach to cutting back on `settings` attributes and replace them by REST API stuff one by one (à la https://github.com/WordPress/gutenberg/pull/23566), but we can probably do that with `initialize` already :man_shrugging:

**Pull request closed without merging**